redbot/core/modlog.py

```diff
@@ -763,6 +767,7 @@ def from_json(cls, name: str, data: dict, **kwargs):
         message=None,
     )
     await _conf.custom(_CASES, str(guild.id), str(next_case_number)).set(case.to_json())
```

Comment:
This needs to be inside a lock on the value for case numbers. We are fetching, awaiting something else, then incrementing here.

Reply:
Good point